server/services/action.js

```diff
@@ -204,6 +208,25 @@ const actions = {
                         case actionTypes.ADD_CASE_NOTE:
                             await caseworkService.post(`/case/${caseId}/note`, { type: 'MANUAL', text: form.data['case-note'] }, headers);
                             break;
+                        case actionTypes.APPLY_CASE_DEADLINE_EXTENSION: {
+                            if (form.data.shouldApplyExtension === 'true') {
+                                const response =
+                                    await caseworkService.post(`/case/${caseId}/stage/${stageId}/extension`,
+                                        { type: form.data.extensionType }, headers);
+
+                                const clientResponse = {
+                                    'summary': `Case: ${response.data.reference} extended`,
+                                    'link': `${response.data.reference}`
+                                };
+
+                                await caseworkService.post(`/case/${caseId}/note`,
```

Comment:
Could the case note be added during the call to /case/${caseId}/stage/${stageId}/extension ?

Reply:
I've updated the applyExtension endpoint in casework to take a reason and create a note.